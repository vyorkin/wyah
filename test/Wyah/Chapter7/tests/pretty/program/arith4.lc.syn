Program [Decl "ten" (ELam (Var "x") (ELam (Var "f") (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EVar (Var "x")))))))))))))),Decl "nine" (ELam (Var "x") (ELam (Var "f") (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EVar (Var "x"))))))))))))),Decl "eight" (ELam (Var "x") (ELam (Var "f") (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EVar (Var "x")))))))))))),Decl "seven" (ELam (Var "x") (ELam (Var "f") (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EVar (Var "x"))))))))))),Decl "six" (ELam (Var "x") (ELam (Var "f") (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EVar (Var "x")))))))))),Decl "five" (ELam (Var "x") (ELam (Var "f") (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EVar (Var "x"))))))))),Decl "four" (ELam (Var "x") (ELam (Var "f") (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EVar (Var "x")))))))),Decl "three" (ELam (Var "x") (ELam (Var "f") (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EVar (Var "x"))))))),Decl "two" (ELam (Var "x") (ELam (Var "f") (EApp (EVar (Var "f")) (EApp (EVar (Var "f")) (EVar (Var "x")))))),Decl "one" (ELam (Var "x") (ELam (Var "f") (EApp (EVar (Var "f")) (EVar (Var "x"))))),Decl "zero" (ELam (Var "x") (ELam (Var "f") (EVar (Var "x"))))]
