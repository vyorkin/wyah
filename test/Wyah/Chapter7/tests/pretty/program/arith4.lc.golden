let ten = \x -> \f -> f (f (f (f (f (f (f (f (f (f x)))))))));
let nine = \x -> \f -> f (f (f (f (f (f (f (f (f x))))))));
let eight = \x -> \f -> f (f (f (f (f (f (f (f x)))))));
let seven = \x -> \f -> f (f (f (f (f (f (f x))))));
let six = \x -> \f -> f (f (f (f (f (f x)))));
let five = \x -> \f -> f (f (f (f (f x))));
let four = \x -> \f -> f (f (f (f x)));
let three = \x -> \f -> f (f (f x));
let two = \x -> \f -> f (f x);
let one = \x -> \f -> f x;
let zero = \x -> \f -> x;